__tests__/users.ts

```diff
@@ -2252,3 +2294,189 @@ describe('mutation unsubscribePersonalizedDigest', () => {
     expect(res.errors).toBeFalsy();
   });
 });
+
+describe('mutation acceptFeatureInvite', () => {
+  const MUTATION = `mutation AcceptFeatureInvite($token: String!, $referrerId: ID!, $feature: String!) {
+    acceptFeatureInvite(token: $token, referrerId: $referrerId, feature: $feature) {
+      _
+    }
+  }`;
+
+  beforeEach(async () => {
+    await con.getRepository(Invite).save({
+      userId: '2',
+      campaign: CampaignType.Search,
+      limit: 5,
+      count: 1,
+      token: 'foo',
+    });
+  });
+
+  it('should require authentication', async () => {
+    await testQueryErrorCode(
+      client,
+      {
+        query: MUTATION,
+        variables: {
+          token: 'foo',
+          referrerId: 2,
+          feature: CampaignType.Search,
+        },
+      },
+      'UNAUTHENTICATED',
+    );
+  });
+
+  it('should return a 404 if the token and referrer mismatch', async () => {
+    loggedUser = '1';
+
+    await testQueryErrorCode(
+      client,
+      {
+        query: MUTATION,
+        variables: {
+          token: 'foo',
+          referrerId: 1,
+          feature: CampaignType.Search,
+        },
+      },
+      'NOT_FOUND',
+    );
+  });
+
+  it('should raise a validation error if the invite limit has been reached', async () => {
+    loggedUser = '1';
+
+    await con
+      .getRepository(Invite)
+      .update({ token: 'foo' }, { limit: 5, count: 5 });
+
+    await testQueryError(
+      client,
+      {
+        query: MUTATION,
+        variables: {
+          token: 'foo',
+          referrerId: 2,
+          feature: CampaignType.Search,
+        },
+      },
+      (errors) => {
+        expect(errors[0].extensions.code).toEqual('GRAPHQL_VALIDATION_FAILED');
+        expect(errors[0].message).toEqual('Invites limit reached');
```

Comment:
Feels like this will need special handling on the client side, is there an internal enum for error codes which we can use for cases such as this?

Reply:
Good point!
@sshanzel are we covering this FE wise?

Dragos: Check the `errors.ts` file, we can leverage that for this part.
Possibly you'll need to introduce a new one.

Reply:
We do have some sort of function to get the error code. We can maybe map the error code or use the message.

Reply:
@rebelchris yes, there is a TODO that says, you should display a toast based on the error message.